indexing-hadoop/src/main/java/io/druid/indexer/hadoop/DatasourceInputFormat.java

```diff
@@ -89,9 +105,11 @@ public int compare(WindowedDataSegment s1, WindowedDataSegment s2)
     List<WindowedDataSegment> list = new ArrayList<>();
     long size = 0;
 
+    JobConf dummyConf = new JobConf();
+    org.apache.hadoop.mapred.InputFormat fio = supplier.get();
```

Comment:
both these lines can be moved inside toDataSourceSplit() itself.

Reply:
> JobConf is rather a big object and I wanted to reuse it if possible.

How come is it a big object? You literally create it completely empty.